src/client/dfuse/dfuse_core.c

```diff
@@ -801,12 +801,6 @@ dfuse_fs_init(struct dfuse_info *dfuse_info,
 
 	fs_handle->dpi_info = dfuse_info;
 
-	/* Max read and max write are handled differently because of the way
-	 * the interception library handles reads vs writes
-	 */
-	fs_handle->dpi_max_read = 1024 * 1024 * 4;
-	fs_handle->dpi_max_write = 1024 * 1024;
```

Comment:
To avoid the fallback to 4K, shall we still set it on older kernels based on proto_major/minor value or any other means?

Reply:
proto is the kernel version, the issue here is the userspace version.  I can't see a foolproof way of always getting what we need, libfuse has gone through several iterations of how to configure buffer size, with both BIG_WRITES and MAX_PAGES being used at different times.  Even now libfuse is documented to use the maximum size for writes by default, but it says nothing for writes.

My preferred way forward is to move to using our own fuse3, but for now it may be a case of landing this and manually testing that the I/O sizes are sensible on the distros that we care about.  This will need testing with both the kernel and userspace for these distros, so cannot be done in docker.